### ISSUE TITLE
Tab Does Not Show Autocomplete with ls Command

### ISSUE THREAD

**Issue by jrock2004** [eza-community/eza#964]
In Mac OS 14.4.1 running era 0.18.14, when I type `ls in` and hit tab, I do not see any autocomplete of directory options. I do not believe this should be. If I type `eza in` and hit tab it autocompletes just fine. If there is multiple matches it brings up to where I can select. 

If I do `which ls` I see the following:

```
ls: aliased to eza --group-directories-first --icons
```

I thought I saw a discussion or a bug that said the placement of the `icons` arg could not be last. If I type the following and hit tab in the terminal the bug shows itself

```
eza --group-directories-first --icons in
```

**Comment by okaufmann** [comment]
I have the exact same probem using iTerm, Spaceship and Zsh with Oh-My-Zsh... Did you manage to fix it yet?

**Comment by jrock2004** [comment]
> I have the exact same probem using iTerm, Spaceship and Zsh with Oh-My-Zsh... Did you manage to fix it yet?

I worked around it by making me own alias

**Comment by Rikaisan** [comment]
> I have the exact same probem using iTerm, Spaceship and Zsh with Oh-My-Zsh... Did you manage to fix it yet?

I was having this same issue, Zsh, Eza, OMZ and Starship, autocomplete was broken when aliased to `ls`.
Turned out my alias was `alias ls="eza -lh --icons=auto --color=auto"`, with the problem being the `-lh`. I was used to having to pass `-h` to see human readable sizes and added that as a habit when aliasing, but since in eza the flag doesn't exist and therefore is invalid, the autocomplete wouldn't work, as soon as I removed the `h`, it started working again.

In short: make sure your alias is valid, check `eza --help` to see all available flags.

**Comment by okaufmann** [comment]
`-h` is a valid option.

From the output of `eza --help`:

```
LONG VIEW OPTIONS
...
  -h, --header               add a header row to each column
```

I am using version `0.18.14`.

**Comment by okaufmann** [comment]
And I found out that `--icons` (without a value) was the problem.

My current alias is working as expected:
```
alias ls="eza -lah --git --icons=always $@"
```

**Comment by cafkafk** [comment]
Seems like the issue was solved :D

**Comment by J-Siu** [comment]
With `zsh` and `ls` alias to `eza -GHF`, I have to use following in my source file:
```
alias ls="eza -GHF \$@"
```

**Comment by d1gitalrain** [comment]
> With `zsh` and `ls` alias to `eza -GHF`, I have to use following in my source file:
> 
> ```
> alias ls="eza -GHF \$@"
> ```

Old thread, but the `-F` flag seems to cause this issue too. As an alternative, using the longform seems to work without needing to escape, eg.
```
alias ls="eza -GH --classify=always"
```

**Comment by jdhao** [comment]
Same issue with with zsh on macOS 15.6.1. The `-F` option somehow prevents tab completion, using the `--classify=always` otpion works.
